.github/workflows/run-tests.yml

```
@@ -0,0 +1,76 @@
+name: Keycloak Client Tests
+
+on:
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  client-tests:
+    name: Client tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        keycloak_server_version: [ "24.0", "26.0", "26.1", "nightly" ]
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: test-setup
+        name: Test setup
+        uses: ./.github/actions/test-setup
+
+      - name: Run client tests
+        run: |
+          mvn -B -f testsuite/providers/pom.xml package -DskipTests=true
+          mvn -B -f testsuite/admin-client-tests/pom.xml test -Dkeycloak.version.docker.image=${{ matrix.keycloak_server_version }}
+          mvn -B -f testsuite/authz-tests/pom.xml test -Dkeycloak.version.docker.image=${{ matrix.keycloak_server_version }}
+
+  client-tests-java-11:
+    name: Client tests (Java 11)
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      matrix:
+        keycloak_server_version: [ "nightly" ]
+    steps:
+      - uses: actions/checkout@v4
+
+      - id: test-setup
+        name: Test setup
+        uses: ./.github/actions/test-setup
+        with:
+          java-version: |
+            11
+            21
```

Multiple versions of Java can be specified here? How does that work? Should this not be exclusively 11?
```suggestion
          java-version: 11
```

---

We need the version 21 to create the providers for the server in the first command (because those providers are for the keytcloak server and it cannot be compiled using java-11).:

```bash
mvn -B -f testsuite/providers/pom.xml package -DskipTests=true
```

---

So the `java-version` can be overloaded into multiple versions? TIL

---

Yes, GHA allows to install several jdks and then there are env vars like `JAVA_HOME_11_X64` available. Check how the maven is started in the worflow for the jdk-11 test. The general `JAVA_HOME` is the last one (21 in this case).